### PR TITLE
feat(internal): docker bundle

### PR DESCRIPTION
**Pull request by viceice**
This pr bundles renovate to a single js file for the docker images

### CODE REVIEW

**Review by JamieMagee — DISMISSED** [review]

**Review by rarkins — COMMENTED** [review]
So this will build Renovate into a single file for the Docker image only? i.e. does not change what's published to to `npmjs`?

File: Dockerfile [hunk 1]
```diff
@@ -210,13 +209,12 @@ FROM $IMAGE as final
...
-COPY --from=tsbuild /usr/src/app/node_modules node_modules
 COPY bin bin
 COPY data data
 
 
 # Numeric user ID for the ubuntu user. Used to indicate a non-root user to OpenShift
 USER 1000
 
-ENTRYPOINT ["node", "/usr/src/app/dist/renovate.js"]
+ENTRYPOINT ["node", "/usr/src/app/dist/index.js"]
```
[the remark this answers is not in the document]
Comment (viceice) [reply Dockerfile:219]: `ncc` always outputs an `index.js` file, that why i create a simple `renovate.js` for backwards compatability

File: lib/util/index.ts [hunk 2]
```diff
@@ -8,3 +10,16 @@ export async function setUtilConfig(
   await setExecConfig(config);
   setFsConfig(config);
 }
+
+/**
+ * Resolve path for a file relative to renovate root directory (our package.json)
+ * @param file a file to resolve
+ */
+export async function resolveFile(file: string): Promise<string> {
```
[the remark this answers is not in the document]
Comment (viceice) [reply lib/util/index.ts:18]: currently is is called 2 times per run, can be more on app long running (travis node shedule)

[the remark this answers is not in the document]
Comment (viceice) [reply lib/util/index.ts:18]: it makes sense to cache the `package.json` path in module scope, so we only do the `findUp` onetime. I'll fix that

**Review by JamieMagee — APPROVED** [review]

**Comment by viceice** [comment]
superseeded by #6009

**Pull request closed without merging**